packages/mui-material/src/Alert/Alert.js

```diff
@@ -52,28 +52,34 @@ const AlertRoot = styled(Paper, {
     padding: '6px 16px',
     ...(color &&
       ownerState.variant === 'standard' && {
-        color: getColor(theme.palette[color].light, 0.6),
-        backgroundColor: getBackgroundColor(theme.palette[color].light, 0.9),
+        color: getColor((theme.vars || theme).palette[color].light, 0.6),
+        backgroundColor: getBackgroundColor((theme.vars || theme).palette[color].light, 0.9),
         [`& .${alertClasses.icon}`]: {
           color:
-            theme.palette.mode === 'dark' ? theme.palette[color].main : theme.palette[color].light,
+            (theme.vars || theme).palette.mode === 'dark'
+              ? (theme.vars || theme).palette[color].main
+              : (theme.vars || theme).palette[color].light,
         },
       }),
     ...(color &&
       ownerState.variant === 'outlined' && {
-        color: getColor(theme.palette[color].light, 0.6),
-        border: `1px solid ${theme.palette[color].light}`,
+        color: getColor((theme.vars || theme).palette[color].light, 0.6),
+        border: `1px solid ${(theme.vars || theme).palette[color].light}`,
         [`& .${alertClasses.icon}`]: {
           color:
-            theme.palette.mode === 'dark' ? theme.palette[color].main : theme.palette[color].light,
+            (theme.vars || theme).palette.mode === 'dark'
+              ? (theme.vars || theme).palette[color].main
+              : (theme.vars || theme).palette[color].light,
         },
       }),
     ...(color &&
       ownerState.variant === 'filled' && {
         color: '#fff',
```

Comment:
Would be great if we consider https://github.com/mui/material-ui/pull/29813 too as part of this PR